src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,123 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        @SuppressWarnings({"unchecked"})
+        Node<K,V>[] newTab = (Node<K,V>[])new Node[DEFAULT_INITIAL_CAPACITY];
+        table = newTab;
+        threshold = (int)(DEFAULT_LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+    }
+
+    private static class Node<K,V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(int hash, K key, V value, Node<K,V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K,V> node = placeNode(key, value);
+        if (node != null) { // existing mapping for key
+            node.value = value;
+            return;
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K,V> e;
+        return (e = getNode(key)) == null ? null : e.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K,V> getNode(K key) {
+        Node<K,V> firstNote = table[(table.length - 1) & hash(key)];
+        Node<K,V> nextNote;
+        int hash = hash(key);
+        if (firstNote != null) {
+            if (firstNote.hash == hash // always check first node
```

Comment:
you definitely copied most of this logic from the source code of the real HashMap, so rewrite all your logic so it looked like yours, because right now it's very difficult to read it and review

Reply:
Not implemented.